### PR TITLE
Custom timezone for built-in timestamp

### LINKED ISSUE #1072: Built-in Variable timestamp: support specifying a time zone
Current: [the variable](https://markbind.org/userGuide/reusingContents.html#built-in-variable-timestamp) is always in GMT

Suggestion: give a way to output the timestamp in a specific timezone

Rationale: If the website is specific to a location e.g., our own module websites, it makes more sense to use the local time in the timestamp

### PR DESCRIPTION
**Pull request by yash-chowdhary**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Enhancement to an existing feature

Fixes #1072 

**What is the rationale for this request?**
Allows custom time zones to be used to generate the `timestamp`. Region-specific websites have a use for this feature, as it doesn't always make sense to refer to UTC/GMT time.

**What changes did you make? (Give an overview)**
* Add support for `"timezone"` and `"locale"` params in `site.json`.
* The 2 params above default to `UTC` and `en-GB`. 

**Is there anything you'd like reviewers to focus on?**
N/A

**Testing instructions:**
1. Add the following to `docs/site.json` -
    ```js
    {
    ...
    "timeZone": "Asia/Singapore",
    "locale": "en-US",
    ...
    }
    ```
1. Scroll down and observe the footer. It should show the local Singapore time in the format - 
  `"<Day (short)>, <Month (short)> <Date>, <Year (long)>, <Time> <AM/PM> <Time Zone>"`
  e.g. `"Generated by MarkBind 2.11.0 on Sun, Mar 1, 2020, 10:14:15 PM GMT+8"` 

**Proposed commit message: (wrap lines at 72 characters)**
Support custom timezones for built-in timestamp variable

### CODE REVIEW

File: docs/userGuide/syntax/variables.mbdf [hunk 1]
```diff
@@ -56,7 +56,21 @@ Represents the root directory of the project. Used for specifying intra-site lin
 
 ##### Built-in Variable: `timestamp`
 
-<code>{<span></span>{ timestamp }}</code> is the time stamp (in UTC) that indicates when the page was generated.
+<code>{<span></span>{ timestamp }}</code> is the time stamp that indicates when the page was generated.
+
+You can set the time zone and locale in `site.json` like so -
```
Comment (ang-zeyu) [inline docs/userGuide/syntax/variables.mbdf]: Should this be moved to the site configuration page instead like the other options?
I'm personally not so familiar with the possible options for each field; Perhaps the format/standard being used here could be stated as well.

Comment (yash-chowdhary) [reply docs/userGuide/syntax/variables.mbdf]: Yes, you're right. It makes more sense for it to be in the configuration page. 
I'll link the possible options for both and explain the date and time format.

Comment (yash-chowdhary) [reply docs/userGuide/syntax/variables.mbdf]: On second thought, following the example of the `enableSearch` parameter, could we reference the `variables.mbdf` in `siteConfiguration.md` to explain the `timeZone` and `locale` parameters?

Comment (ang-zeyu) [reply docs/userGuide/syntax/variables.mbdf]: Sounds good as well

File: docs/userGuide/syntax/variables.mbdf [hunk 2]
```diff
@@ -58,19 +58,9 @@ Represents the root directory of the project. Used for specifying intra-site lin
...
-```js
-site.json
-{
-  ...
-  "timeZone": "Asia/Singapore",
-  "locale": "en-US",
-  ...
-}
-```
+Compiled in the following gists are lists of possible [time zone](https://gist.github.com/yash-chowdhary/136df3f6721f2fd1d5ea46b14f7c531c) and [locale](https://gist.github.com/yash-chowdhary/674654bdd977b063de008c1b3655a515) options.
```
Comment (ang-zeyu) [inline docs/userGuide/syntax/variables.mbdf]: Hmm why not add it into the docs itself?
Perhaps a minimal panel or link leading to a new page in the docs

Comment (yash-chowdhary) [reply docs/userGuide/syntax/variables.mbdf]: Added it in a minimal panel 👍

File: docs/userGuide/siteConfiguration.md [hunk 3]
```diff
@@ -198,3 +198,23 @@ The example above uses tags as an example of configuring plugin settings, refer
...
+
+#### **`locale`**
+
+**Language by locale used for the [time stamp](reusingContents.html#built-in-variable-timestamp).** Default: `"en-GB"` (`English (United Kingdom)`). <br>
+The date format is thus - <br>
+`<Day>, <Date> <Month> <Year>, <24-hour Time> <Time Zone Code>`. 
+
+<panel type="minimal" header="Locale Options">
+  <include src="pages/locales.md" />
+</panel>
```
Comment (ang-zeyu) [inline docs/userGuide/siteConfiguration.md]: Just one last nit: Maybe add a `<br>` after this as its a little tight

**Comment by damithc** [comment]
Works in production. Nice work @yash-chowdhary and reviewers. 👍

**Pull request merged**